IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -235,126 +217,132 @@ - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
             }
         }
     }
+    
+    [self fetchCachedRefreshTokenWithCompletionHandler:^(MSIDBaseToken<MSIDRefreshableToken> *refreshToken, MSIDRefreshTokenTypes tokenType, NSError *error) {
+        if (error)
+        {
+            completionBlock(nil, error);
+            return;
+        }
+        
+        [self tryRefreshToken:refreshToken tokenType:tokenType completionBlock:completionBlock];
+    }];
+    
+}
 
-    NSError *frtCacheError = nil;
-    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Looking for Family Refresh token...");
-    MSIDRefreshToken *familyRefreshToken = [self familyRefreshTokenWithError:&frtCacheError];
-
-    if (frtCacheError)
-    {
-        MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, self.requestParameters, @"Failed to read family refresh token with error %@", MSID_PII_LOG_MASKABLE(frtCacheError));
-        completionBlock(nil, frtCacheError);
-        return;
-    }
-
-    if (familyRefreshToken)
+- (void)fetchCachedRefreshTokenWithCompletionHandler:(void (^)(MSIDBaseToken<MSIDRefreshableToken> *, MSIDRefreshTokenTypes, NSError *))completionHandler
+{
+    if (!completionHandler)
     {
-        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Found Family Refresh token, using it...");
-        [self tryFRT:familyRefreshToken completionBlock:completionBlock];
+        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"No completionHandler when fetch local refresh token");
```

Comment:
nit: I think you should just return early here :) That would avoid an else condition and calling nil completion block on line 286

Reply:
Updated as suggested